src/API/Reports/Customers/DataStore.php

```diff
@@ -484,6 +524,29 @@ public static function get_or_create_customer_from_order( $order ) {
 			return $returning_customer_id;
 		}
 
+		list($data, $format) = self::get_customer_order_data_and_format( $order );
+
+		$result      = $wpdb->insert( self::get_db_table_name(), $data, $format );
+		$customer_id = $wpdb->insert_id;
+
+		/**
+		 * Fires when a new report customer is created.
+		 *
+		 * @param int $customer_id Customer ID.
+		 */
+		do_action( 'woocommerce_analytics_new_customer', $customer_id );
+
+		return $result ? $customer_id : false;
+	}
+
+	/**
+	 * Returns a data object and format object of the customers data coming from the order.
+	 *
+	 * @param object      $order         WC_Order where we get customer info from.
+	 * @param object|null $customer_user WC_Customer registered customer WP user.
+	 * @return array ($data, $format)
+	 */
+	public static function get_customer_order_data_and_format( $order, $customer_user = null ) {
```

Comment:
This was overlapping logic between `sync_order_customer` and `get_or_create_customer_from_order`